### DISCUSSION TITLE
Submitting result of one form into another

### DISCUSSION THREAD

**Discussion by NavaceSystem** [react-hook-form/react-hook-form#10268] [Q&A] [answered]
Check this repro: https://codesandbox.io/s/react-hook-form-reset-v7-ts-forked-q8pi2r?file=/src/index.tsx

I have a main form which has `items`, which is an Array of `{ firstName, lastName }`. I'm trying to have a separate temporary form with only firstName and lastName which when submitted, will append to the main array form and reset it's own state. The issue is I can't even get the value to submit correctly. It has something to do with the handleSubmit of the smaller form, if I use a simple click handler that just does `fieldArray.append({firstName: 'a', lastName:'b')` it works. It just doesn't work through `addItemForm.handleSubmit` handler as shown in the repo.

I must be missing something as this is a fairly common use case where you have an "item form" that's validated and when submitted is dumped into the "array form" and the item form is reset.

Any help appreciated.

**Comment by leapful** [comment, marked as answer]
@NavaceSystem 

Could you please upgrade to latest version of react-hook-form ?

It works fine with v7.43.9: https://codesandbox.io/s/react-hook-form-reset-v7-ts-forked-w798sk?file=/src/index.tsx

**Reply by NavaceSystem** [reply]
Ah got it, thank you.
